### PR TITLE
Fix: pass user configuration to Tool prepare method

### PR DESCRIPTION
**Pull request by gohabereg**

### CODE REVIEW

**Comment by Broutard** [comment]
Please validate this PR which solve https://github.com/editor-js/embed/issues/68

File: src/components/modules/tools.ts [hunk 1]
```diff
@@ -279,7 +279,7 @@ export default class Tools extends Module {
   }[] {
     const toolPreparationList: {
       function: (data: { toolName: string }) => void | Promise<void>;
-      data: { toolName: string };
+      data: { toolName: string; config: ToolConfig };
```
Comment (neSpecc) [inline src/components/modules/tools.ts:282]: need to add this to the function's return type

**Comment by georgeciubotaru** [comment]
@neSpecc  Are you guys planning to fix this issue, or we are chilling now?

**Comment by Broutard** [comment]
It would be really appreciated if this bug could be fixed quickly, please.
We cannot update after 2.19.3 because of this issue.

@gohabereg do you have time to do this ?

**Review by neSpecc — APPROVED** [review]

**Comment by gohabereg** [comment]
resolves editor-js/embed#68
resolves editor-js/embed#60
resolves editor-js/embed#72

**Review by talyguryn — APPROVED** [review]

**Comment by Farrien** [comment]
This fix took 5-6 months LMAO

**Pull request merged**